draft-packages/menu/KaizenDraft/Menu/components/MenuItem.tsx

```diff
@@ -6,15 +6,36 @@ const styles = require("./MenuContent.module.scss")
 
 export type MenuItemProps = {
   label: string
-  action: string | ((e: React.MouseEvent<HTMLAnchorElement>) => void)
+  /**
+   * Historically, we only had `action`, which would accept either a string for
+   * the href, or a function for an onClick callback.
+   * The limitation here was that it prevented us from having
+   * both a href and an onClick callback at the same time.
+   * This field has been deprecated in favour of `onClick` and `href`.
+   * @deprecated
+   */
+  action?: string | ((e: React.MouseEvent<HTMLButtonElement>) => void)
```

Comment:
Great to see this! This union type has been a real pain to deal with in the Title Block.